### DISCUSSION TITLE
Add `selectEntries` as a `mapEntries` that can also filter out entries

### DISCUSSION THREAD

**Discussion by aleclarson** [radashi-org/radashi#57] [Ideas]
Copied from https://github.com/sodiray/radash/issues/364

Basically, it does the following but in a more performant manner:

```ts
import { select } from 'radashi'

export function selectEntries(object, map, filter) {
  return Object.fromEntries(
    select(
      Object.entries(object),
      map,
      filter,
    ),
  )
}

selectEntries(
  object,
  ([key, value]) => [key.toUpperCase(), `${value}`],
  ([key]) => key !== 'power',
)
```

PR welcome

**Comment by Minhir** [comment]
For me, `select` looks like [**Abstraction over sugar**](https://github.com/orgs/radashi-org/discussions/20). But select is already here, so I don't propose to remove it. Just trying to be careful with introducing new API. What do you think about `select` type extension? It can also accept an object as first argument. 

```js
select(
  object,
  ([key, value]) => [key.toUpperCase(), `${value}`],
  ([key]) => key !== 'power',
)
```

**Reply by aleclarson** [reply]
Yeah I think you're right that `selectEntries` isn't gonna cut it. It's better for readability to write out the loop manually:
```ts
const newObj = {}
for (const [key, value] of Object.entries(newObj)) {
  if (condition(key, value)) {
    newObj[key] = value
  }
}
```
